src/main/java/io/vertx/core/http/impl/Http2ServerResponseImpl.java

```diff
@@ -601,7 +601,13 @@ public HttpServerResponse sendFile(String filename, long offset, long length, Ha
           }
         }
         checkSendHeaders(false);
-        file.pipeTo(this, h);
+        file.pipeTo(this, ar1 -> file.close(ar2 -> {
```

Comment:
can you try instead : file.pipe().endOnComplete(true).to(h) this should have the same effect